src/Client/HandlerDelegates.cs

```diff
@@ -0,0 +1,89 @@
+﻿using OmniSharp.Extensions.LanguageServer.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OmniSharp.Extensions.LanguageServerProtocol.Client
+{
+    /// <summary>
+    ///     A handler for empty notifications.
+    /// </summary>
+    /// <returns>
+    ///     A <see cref="Task"/> representing the operation.
+    /// </returns>
+    public delegate void EmptyNotificationHandler();
+
+    /// <summary>
+    ///     A handler for notifications.
+    /// </summary>
+    /// <typeparam name="TNotification">
+    ///     The notification message type.
+    /// </typeparam>
+    /// <param name="notification">
+    ///     The notification message.
+    /// </param>
+    public delegate void NotificationHandler<TNotification>(TNotification notification);
```

Comment:
I recently changed these on the server to be `Task` instead of `void` to allow them to block processing until the operation completes, this ensures things like text updates are completed.

Question: Should we do the same here for the client?  (I honestly don't know)

Reply:
Interesting point; I think that makes sense server-side, but I reckon _sending_ a notification should be fire-and-forget. I can't think of a reason that a client would want to block until it's been sent.

Reply:
and thats why I ask the question. :D